packages/@o3r/extractors/src/core/comparator/metadata-comparator.interface.ts

```diff
@@ -0,0 +1,78 @@
+import type { JsonObject } from '@angular-devkit/core';
+import type { SupportedPackageManagers } from '@o3r/schematics';
+
+/**
+ * Interface of the comparator used to compare 2 different versions of the same metadata file.
+ */
+export interface MetadataComparator<T> {
+  /**
+   * Get an array of metadata items to parse a metadata file content.
+   * @param content Content of a metadata file
+   */
+  getArray(content: any): T[];
+
+  /**
+   * Get a description of a metadata item.
+   * @param item Metadata item
+   */
+  getName(item: T): string;
+
+  /**
+   * Compares 2 metadata items.
+   * @param item1 Metadata item
+   * @param item2 Metadata item to compare with
+   */
+  isSame(item1: T, item2: T): boolean;
+
+  /**
+   * Returns true if a migration item matches a metadata item.
+   * @param metadataItem Metadata item
+   * @param migrationItem Migration item
+   */
+  isMigrationDataMatch(metadataItem: T, migrationItem: any): boolean;
+}
+
+/**
+ * Migration item used to document a migration of a config, localization or styling metadata.
+ */
+export interface MigrationData<T> {
+  /** Metadata type */
```

Comment:
Shouldn't we have a model like 
```typescript 
export interface MigrationData<T> {
  /** Metadata type */
  contentType: string;

  /** Previous metadata value */
  before: T;

  /** New metadata value */
  after: T;
}
```
and have each package defining their interface like this
```typescript
export interface ConfigMigrationData extends MigrationData<ComponentConfigOutput> {
  /** @inheritDoc */
  contentType: 'CONFIG';
}
```